ui/src/app/job-list/job-list.component.ts

```diff
@@ -151,6 +163,15 @@ export class JobListComponent implements OnInit {
     this.jobStream.loadAtLeast((e.pageIndex+2) * e.pageSize)
       .catch((error) => this.handleError(error));
   }
+
+  toggleDisplayColumn(field: DisplayField) {
+    const newValue = !field.showInListView;
+    this.displayFields = this.settingsService.setDisplayColumn(field, newValue);
+  }
+
+  saveSettings() {
+    this.jobTable.ngOnInit();
```

Comment:
I don't think you should call this directly, can you make `setUpFieldsAndColumns` public and just call that?